test/com/google/enterprise/adaptor/database/JdbcFixture.java

```diff
@@ -207,6 +208,17 @@ public static void executeUpdate(String... sqls) throws SQLException {
     }
   }
 
+  public static PreparedStatement prepareStatement(String sql)
+      throws SQLException {
+    Connection conn = getConnection();
+    PreparedStatement stmt = conn.prepareStatement(sql);
+    synchronized (openObjects) {
```

Comment:
Fewer locking/unlocking cycles, but having getConnection() add the connection to openObjects, made me change it to a synchronized collection.